pandas/io/formats/format.py

```diff
@@ -565,315 +583,446 @@ def __init__(
+
+        # Ignoring error
+        # expression has type "Union[List[Callable[..., Any]],
+        # Tuple[Callable[..., Any], ...],
+        # Mapping[Union[str, int], Callable[..., Any]], None]",
+        # variable has type "Union[List[Callable[..., Any]],
+        # Tuple[Callable[..., Any], ...], Mapping[Union[str, int],
+        # Callable[..., Any]]]")
+        self.formatters = formatters  # type: ignore[assignment]
+        self.na_rep = na_rep
+        self.decimal = decimal
 
+        # Ignoring error
+        # expression has type "Union[str, int, Sequence[Union[str, int]],
+        # Mapping[Optional[Hashable], Union[str, int]], None]",
+        # variable has type "Mapping[Optional[Hashable], Union[str, int]]"
+        self.col_space = col_space  # type: ignore[assignment]
+
+        self.header = header
+        self.index = index
+        self.line_width = line_width
+        self.max_rows = max_rows
+        self.min_rows = min_rows
+        self.max_cols = max_cols
+        self.show_dimensions = show_dimensions
+        self.table_id = table_id
+        self.render_links = render_links
+        self.justify = justify
+        self.bold_rows = bold_rows
+        self.escape = escape
+
+        # Ignoring error:
+        # expression has type "Optional[Sequence[str]]", variable has type "Index"
+        self.columns = columns  # type: ignore[assignment]
+
+        self._truncate()
+        self.adj = get_adjustment()
+
+    @property
+    def sparsify(self):
+        return self._sparsify
+
+    @sparsify.setter
+    def sparsify(self, sparsify):
         if sparsify is None:
-            sparsify = get_option("display.multi_sparse")
+            self._sparsify = get_option("display.multi_sparse")
+        else:
+            self._sparsify = sparsify
 
-        self.sparsify = sparsify
+    @property
+    def formatters(self) -> FormattersType:
+        return self._formatters
 
-        self.float_format = float_format
+    @formatters.setter
+    def formatters(self, formatters: Optional[FormattersType]) -> None:
+        self._formatters: FormattersType
         if formatters is None:
-            self.formatters = {}
-        elif len(frame.columns) == len(formatters) or isinstance(formatters, dict):
-            self.formatters = formatters
+            self._formatters = {}
+        elif len(self.frame.columns) == len(formatters) or isinstance(formatters, dict):
+            self._formatters = formatters
         else:
             raise ValueError(
                 f"Formatters length({len(formatters)}) should match "
-                f"DataFrame number of columns({len(frame.columns)})"
+                f"DataFrame number of columns({len(self.frame.columns)})"
             )
-        self.na_rep = na_rep
-        self.decimal = decimal
+        assert self._formatters is not None
+
+    @property
+    def justify(self):
+        return self._justify
+
+    @justify.setter
+    def justify(self, justify):
```

Comment:
are these properties set dynamically (i.e. after `__init__`?)

Reply:
No, turns out that they are used only in ``__init__``.

Reply:
yeah i would be in favor of simply typing these on the class. i am not sure why that is causing issues; we do this in lots of places and its the idiomatic soln. (for something you are setting that is different), but if you are *only* setting it in ``__init__`` then prefer to not have these accessors